protobuf-c/protobuf-c.c

```diff
@@ -348,15 +348,15 @@ uint64_size(uint64_t v)
 
 	if (upper_v == 0) {
 		return uint32_size((uint32_t) v);
-	} else if (upper_v < (1 << 3)) {
+	} else if (upper_v < ((int32_t)1 << 3)) {
 		return 5;
-	} else if (upper_v < (1 << 10)) {
+	} else if (upper_v < ((int32_t)1 << 10)) {
 		return 6;
-	} else if (upper_v < (1 << 17)) {
+	} else if (upper_v < ((int32_t)1 << 17)) {
 		return 7;
-	} else if (upper_v < (1 << 24)) {
+	} else if (upper_v < ((int32_t)1 << 24)) {
 		return 8;
-	} else if (upper_v < (1U << 31)) {
+	} else if (upper_v < ((int32_t)1 << 31)) {
```

Comment:
I don't think this is correct, since a 31 bit shift on a signed 32-bit integer will make the resulting value negative? At the very least this is more confusing. I think the constants should be cast `(uint32_t)` here and in all other changed cases for consistency.